plugins/statistics.koplugin/main.lua

```diff
@@ -280,31 +421,30 @@ function ReaderStatistics:getTotalStats()
            total_stats
 end
 
+
 function ReaderStatistics:getStatisticsFromHistory(total_stats)
     local titles = {}
     local total_books_time = 0
-    for curr_file in lfs.dir(history_dir) do
-        local path = joinPath(history_dir, curr_file)
-        if lfs.attributes(path, "mode") == "file" then
-            local book_result = self:importFromFile(history_dir, curr_file)
-            local book_stats = book_result.stats
-            if book_stats and book_stats.total_time_in_sec > 0
-                    and book_stats.title ~= self.data.title then
-                titles[book_stats.title] = true
-                table.insert(total_stats, {
-                    book_stats.title,
-                    util.secondsToClock(book_stats.total_time_in_sec, false),
-                    callback = function()
-                        UIManager:show(KeyValuePage:new{
-                            title = book_stats.title,
-                            kv_pairs = getDatesForBook(book_stats),
-                        })
-                    end,
-                })
-                total_books_time = total_books_time + tonumber(book_stats.total_time_in_sec)
-            end
-        end
-    end
+    for _, v in pairs(ReadHistory.hist) do
+        local book_settings = DocSettings:open(v.file)
+        local book_stats = book_settings.data.stats
```

Comment:
I suggest change this to:

```lua 
local book_stats = DocSettings:open(v.file):readSetting('stats')
```

Then, we can also get rid of `book_settings` variable.